### ISSUE TITLE
Block request: Taskwarrior

### ISSUE THREAD

**Issue by jcgruenhage** [greshake/i3status-rust#314]
It would be cool to have a taskwarrior block that indicates the current status of your tasks, something like X ready tasks, Y nearly due tasks, Z overdue tasks.

I'd be willing to implement this, but it's not that far up my todo list yet, so if someone else want to do it first, feel free!

**Comment by atheriel** [comment]
It's probably possible to do this with the Custom block already. I've been a bit hesitant to add support for todo managers in the past (see #195), since none of them seem to have particularly widespread adoption. But others may disagree.

**Comment by jcgruenhage** [comment]
If it's behind a feature flag I don't see the harm in it though

**Comment by jubnzv** [comment]
I already have simple taskwarrior block for personal use here: https://github.com/jubnzv/i3status-rust/commit/6dc80849a4270ccf76a0540e9cf512b832efc585. It's implemented as a wrapper around `task` command (following [these](https://taskwarrior.org/docs/3rd-party.html) guidelines), so it may be cleaner use Custom block instead. Something like:

```
[[block]]
block = "custom"
command = "echo  `task rc.gc=off sched.before:today+1d -COMPLETED -DELETED -DUETODAY count`"
interval = 60
```

If it can be useful for someone else, I could extend my wrapper functionality to separate crate and create PR with a new block. Any thoughts?

**Comment by atheriel** [comment]
I tend to think that if a block is simply displaying the output of a shell command every minute, there is not much advantage to a "native" block over using the custom block, as you have in your example.

**Comment by jcgruenhage** [comment]
Can custom blocks do colors? I'd like to have the block red when tasks are overdue, yellow when there's still things due soon or scheduled for today and plain gray otherwise.

**Comment by atheriel** [comment]
Yep! Semantically meaningful colours would be one good reason to use a native block. Adding colour to the count from a shell command is exactly what the Pacman block does, for example.

**Comment by ammgws** [comment]
How about enabling pango markup for the various widgets (button, text, etc)? Then should be able to add colour to any custom block with just the shell output

**Comment by atheriel** [comment]
@ammgws Oh, that's a good point. Have I missed an open issue/PR for this?

**Comment by ammgws** [comment]
I had a PR merged for the separator block, leaving the other blocks for later since I wanted to test it first. I can open one if the test proves promising.

**Comment by ammgws** [comment]
![20190319_22h57m01s](https://user-images.githubusercontent.com/20397027/54611495-687dba80-4a9a-11e9-831d-e6ff44fff64b.png)
Seems to work fine so I'll open a PR.

**Comment by jcgruenhage** [comment]
That is just the text color, not the block color though, right?

**Comment by ammgws** [comment]
You can set the block colour too. See the second image in #346

**Comment by jcgruenhage** [comment]
That is the text background colour, not the block colour.

**Comment by ammgws** [comment]
Implemented via #600
